### PR TITLE
Conversion of node-sass to dart-sass and enabling SASS MODULES

### PR DESCRIPTION
**Pull request by illepic**
Sass modules will help us drastically simplify our Sass imports (no more printing/non-printing concepts).

https://css-tricks.com/introducing-sass-modules/

### CODE REVIEW

**Comment by illepic** [comment]
Something we have to figure out how to solve:

`@use` is wonderful for reducing duplicated output ... in a single Sass project. Unfortunately, every time Webpack sees us bring in a `.scss file`, it's essentially a brand new Sass project instance. There is no shared "state". This means that we're currently not getting any of the benefits of `@use` within our components.

Here is the issue on the official Sass project: https://github.com/sass/sass/issues/2745

**Comment by illepic** [comment]
Aight, we're addressing this with the similar "non-printing" approach, this time behind the `utils` namespace. For examples, Take a look at:

- https://github.com/phase2/particle/pull/628/files#diff-d0ac9ae867c6e963d2a8196651e7cc7d
- https://github.com/phase2/particle/pull/628/files#diff-ad41726267defa7e198efeea4dbd0f16

This should be a lot clearer going forward.

File: apps/node-pl/webpack.config.js [hunk 1]
```diff
@@ -6,10 +6,11 @@
 const path = require('path');
 
 const { DefinePlugin } = require('webpack');
-const sassExportData = require('@theme-tools/sass-export-data');
+// const sassExportData = require('@theme-tools/sass-export-data');
```
Comment (dslemay) [inline apps/node-pl/webpack.config.js]: Should this comment be removed?

File: source/default/clientside/vue-widget/src/vue-cards/components/banner.vue [hunk 2]
```diff
@@ -61,18 +61,19 @@ export default {
...
-// Sass here.
 .vue-banner {
-  margin: map-get($spacers, 5);
-  padding: map-get($spacers, 4);
-  font-size: $font-size-lg;
-  background-color: map-get($theme-colors, 'new');
+  margin: map.get(utils.$bs-spacers, 5);
+  padding: map.get(utils.$bs-spacers, 4);
+  font-size: utils.$bs-font-size-lg;
+  // background-color: mapget($theme-colors, 'new');
```
Comment (dslemay) [inline source/default/clientside/vue-widget/src/vue-cards/components/banner.vue]: Should this comment be removed?

File: apps/node-pl/pattern-lab/_data/sass-variables.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"scssColors":{"primary":"rgba(0,123,2551)","secondary":"rgba(108,117,1251)","success":"rgba(40,167,691)","info":"rgba(23,162,1841)","warning":"rgba(255,193,71)","danger":"rgba(220,53,691)","light":"rgba(244,244,2441)","dark":"rgba(52,58,641)","pink":"rgba(255,192,2031)","goldenrod":"rgba(218,165,321)"},"scssBreakpoints":{"xs":0,"sm":"576px","md":"768px","lg":"992px","xl":"1200px"},"scssSpacing":{"0":0,"1":"0.25rem","2":"0.5rem","3":"1rem","4":"1.5rem","5":"3rem","6":"4rem"},"fontFamilies":{"$font-family-sans-serif":"Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif","$font-family-monospace":"Source Code Pro, Courier New, monospace","$headings-font-family":"Oswald, Times New Roman, serif"},"fontSizes":{"base":"1rem","lg":"1.25rem","sm":"0.875rem"},"fontWeights":{"lighter":"lighter","light":300,"normal":400,"bold":700,"bolder":"bolder"},"scssIcons":["chevron-down","chevron-left","chevron-right","chevron-up","close","menu","person","search"]}
```
[the remark this answers is not in the document]
Comment (illepic) [reply apps/node-pl/pattern-lab/_data/sass-variables.json]: It's just robot food, but definitely has to exist prior to a build.

File: apps/node-pl/scss/_scss2json.scss [hunk 4]
```diff
@@ -1,50 +1,47 @@
...
+@use 'sass:map';
+// Custom
+@use '~tokens/sass/utils';
+@use '~atoms/svg/svg';
 
-// Customized Sass Mapping
-// Pulled from Bootstrap's _variables.scss
 $font-families-demo: (
-  '$font-family-sans-serif': $font-family-sans-serif,
-  '$font-family-serif': $font-family-serif,
```
[the remark this answers is not in the document]
Comment (illepic) [reply apps/node-pl/scss/_scss2json.scss]: Became `$headings-font-family`.

**Comment by illepic** [comment]
@dslemay where are you seeing the comments still existing about prepending of the magic Sass?

**Comment by dslemay** [comment]
@illepic Disregard on the prepending of magic sass. Missed it being removed from the Yeoman file. Looks like it is also removed there. https://github.com/phase2/particle/blob/feature/sass-modules/tools/generators/new-component/generators/app/templates/ds/_pattern.scss.ejs

File: apps/node-pl/pattern-lab/_data/sass-variables.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (dslemay) [reply apps/node-pl/pattern-lab/_data/sass-variables.json]: Makes sense. 👍

**Pull request merged**